codec/src/main/java/io/netty/handler/codec/DefaultHeaders.java

```diff
@@ -104,16 +172,21 @@ public T getAndRemove(T name, T defaultValue) {
     }
 
     @Override
-    @SuppressWarnings("unchecked")
     public List<T> getAll(T name) {
-        Object value = map.get(name);
-        if (isList(value)) {
-            return unmodifiableList((List<T>) value);
-        }
-        if (value == null) {
-            return emptyList();
+        checkNotNull(name, "name");
+
+        LinkedList<T> values = new LinkedList<T>();
```

Comment:
@Scottmitch ic. 1) why do we have to do `addFirst`? 2) can we add a comment to that's the reason why LinkedList?